src/components/Text/src/Text.vue

```diff
@@ -123,21 +132,35 @@ export default {
 		},
 		inlineStyles() {
 			const { fonts } = this.theme;
+
 			return {
-				fontFamily: this.resolvedFontFamily,
-				fontWeight: this.resolvedFontWeight,
+				fontFamily: this.useThemeComponentData ? this.resolvedFontFamily : this.fontFamily,
+				fontWeight: this.useThemeComponentData ? this.resolvedFontWeight : this.fontWeight,
 				color: this.resolvedColor,
 				fontSize: this.fontSize,
 				lineHeight: this.lineHeight,
 				'--mobile-base-font-size': fonts.baseSize,
 				'--mobile-font-size-scale': fonts.sizeScale,
 			};
 		},
+		/*
```

Comment:
I don't think this does anything. Once we remove `theme.text` none of the theme component data will be applied here. They'll all be applied on MTheme as css and referenced in Text.vue

Reply:
I didn't want to imply that we should remove `theme.text`. I think we should continue to support the ability to pass data to components the same way we do now and expand on that with variants.

This is more meant to be a short-term approach to provide that data that shouldn't change the public API. I think both ways of passing data would work, but I'd like to spend more time to evaluate how reusable css is generated and how the variants would work together.